test/test_response.py

```diff
@@ -448,6 +448,49 @@ def test_decode_zstd_incomplete(self, data: bytes) -> None:
         with pytest.raises(DecodeError):
             HTTPResponse(fp, headers={"content-encoding": "zstd"})
 
+    decode_param_set = [
+        b"f",
+        b"fo",
+        b"foo",
+        b"foo" * 5,
+        b"123456789012345",
+        b"foo4",
+        b"x" * 100,
+    ]
+
+    @onlyZstd()
+    @pytest.mark.parametrize("data", decode_param_set)
+    def test_decode_zstd_incomplete_read(self, data: bytes) -> None:
+        data = zstd.compress(data)
+        fp = BytesIO(data[:-1])  # shorten the data to trigger DecodeError
+
+        # create response object without(!) reading/decoding the content
+        r = HTTPResponse(
+            fp, headers={"content-encoding": "zstd"}, preload_content=False
+        )
+
+        # read/decode, expecting DecodeError
+        with pytest.raises(DecodeError):
+            r.read(decode_content=True)
+
+    @onlyZstd()
+    @pytest.mark.parametrize("data", decode_param_set)
+    def test_decode_zstd_incomplete_read1(self, data: bytes) -> None:
+        data = zstd.compress(data)
+        fp = BytesIO(data[:-1])
+
+        r = HTTPResponse(
+            fp, headers={"content-encoding": "zstd"}, preload_content=False
+        )
+
+        # read/decode via read1(!), expecting DecodeError
+        with pytest.raises(DecodeError):
+            amt_decoded = 0
+            # loop, as read1() may return just partial data
+            while amt_decoded < len(data):
+                part = r.read1(decode_content=True)
+                amt_decoded += len(part)
```

Comment:
What do you think about making the behavior check more explicit by checking that the first `read1` returns the truncated data and the second one leads to `DecodeError`? Reading all data from a not changing `BytesIO` object always requires just one `read1()` unless the amount is specified.

Reply:
me bit dizzy from the other issues to understand. If the test-case is valid, it should go in, can be extended/refined later.

(note that this is in context of https://github.com/urllib3/urllib3/issues/3250, so while I get a better understanding, the teest-cases will be refined anyways)

Reply:
=> after a bit thought: keep test focused on DecodeError